src/modules/emote_menu/components/Menu.jsx

```diff
@@ -0,0 +1,47 @@
+import React, {useEffect, useState} from 'react';
+import Divider from 'rsuite/lib/Divider/index.js';
+import emoteStore from '../stores/index.js';
+import styles from '../styles/menu.module.css';
+import Emotes from './Emotes.jsx';
+import Header from './Header.jsx';
+import Preview from './Preview.jsx';
+import Sidebar from './Sidebar.jsx';
+
+export default function EmoteMenu({triggerRef}) {
+  const [search, setSearch] = useState('');
+  const [emote, setEmote] = useState(null);
+  const [focus, setFocus] = useState({
+    eventKey: emoteStore.getHeaders()[0].id,
+    scrollTo: false,
+  });
+
+  const onHide = () => triggerRef.current.close();
+
+  useEffect(() => setSearch(''), [focus]);
+
+  function handleClick(newEmote) {}
```

Comment:
i assume this is a TODO which will insert into the message input?

Reply:
Yes, and handle toggling favorites with `alt` and keeping the menu open with `shift`